internal/cmd/connect/install_utils.go

```diff
@@ -164,35 +165,16 @@ func hasArchiveInstallation(dir string) bool {
 }
 
 func compactDuplicateInstallations(installations []installation) []installation {
-	if len(installations) == 0 {
-		return []installation{}
-	}
-
-	sort.SliceStable(installations, func(i, j int) bool {
-		if installations[i].Type == installations[j].Type {
-			return installations[i].Path < installations[j].Path
-		}
-		return installations[i].Type < installations[j].Type
-	})
-
-	equalInstallations := func(i, j int) bool {
-		return (installations[i].Type == installations[j].Type) && (installations[i].Path == installations[j].Path)
-	}
-
 	var uniqueInstallations []installation
-	i := 0
-	for j, ins := range installations {
-		if j == i {
-			continue
-		}
-		if equalInstallations(i, j) {
-			installations[i].Use = fmt.Sprintf("%s, %s", installations[i].Use, ins.Use)
-		} else {
-			uniqueInstallations = append(uniqueInstallations, installations[i])
-			i = j
+
+	s := types.NewSet()
+	for _, ins := range installations {
+		typePathStr := fmt.Sprintf("type=%s,path=%s", ins.Type, ins.Path)
```

Comment:
Is there anything we can do to make `set` generic for any struct, so we don't have to do this?

Reply:
I'm pretty sure we can do something with generics. Should be a simple followup.

Reply:
This ended up being pretty simple and didn't require too many changes elsewhere in the code, so I'll roll it into this PR.

Since I'm only comparing two of the fields of this struct, I modified the struct as well.